### PR TITLE
Update dev/from release

### LINKED ISSUE #871: Some New Endpoints Return Sensitive Data - Should Require Auth
The following are endpoints that I tested and that, for some reason or another (REGI?), have IP address/User Information/Hostnames in them. 

Would the best method be to force authentication of these GET requests? 

# Project Lock Rights Endpoint

### Command
```bash
curl -X 'GET' \  'https://T7HOST:TCPORT/swt-data/project-lock-rights?office-mask=SWT' \  -H 'accept: application/json'
```

### Result
Having a user in REGI create a lock, then running that to see the output gives this
```json
[
  {
    "office-id": "SWT",
    "project-id": "ALTU",
    "application-id": "regi;daily ops",
    "acquire-time": 1725640446000,
    "session-user": "<USERNAME REDACTED>",
    "os-user": "<USERNAME REDACTED>",
    "session-program": "JDBC Thin Client",
    "session-machine": "<SYSTEM HOSTNAME REDACTED>"
  }
]
```

# Project Lock Revoker Rights Endpoint
### Command
```bash
curl -X 'GET' \
  'https://T7HOST:TCPORT/swt-data/project-lock-rights?office-mask=SWT' \
  -H 'accept: application/json'
  ```
  ### Result
  ```json
  [
  {
    "office-id": "SWT",
    "project-id": "BIGH",
    "application-id": "regi",
    "user-id": "<REDACTED USERID 1>"
  },
  {
    "office-id": "SWT",
    "project-id": "WAUR",
    "application-id": "regi",
    "user-id": "<REDACTED USERID 2>"
  },
  ...
  ]
  ```

### PR DESCRIPTION
**Pull request by MikeNeilson**
I'm intending for this to be the last release branch based build in favor of switching to the git tag based releases we've started using in other projects. Part of that will be making the integration tests part of the merge gate (we still have 6 that are failing but it's known the work isn't done yet.)

That should help keep the number of files reduced any time I do these types of staging.

Good news is I was able to keep up with develop fairly well so these changes are mostly overriding or updating dependencies.. and fixing depedabot.

### CODE REVIEW

**Comment by MikeNeilson** [comment]
NOTE: there is one more change before release. Not sure if I'll do that in this style or try the new style. 

We need to hide the at_properties behind a login.

**Comment by MikeNeilson** [comment]
NOTE: for the TeamCity build, only the expected tests failed. Those endpoints are still in development.

**Comment by MikeNeilson** [comment]
@krowvin is currently gathering if we need to give water supply contracts the same treatment. Once that's determined I'm going to call it a release.

**Review by adamkorynta — DISMISSED** [review]

File: cwms-data-api/src/main/java/cwms/cda/formatters/Formats.java [hunk 1]
```diff
@@ -280,6 +280,30 @@ public static ContentType parseHeaderAndQueryParm(String header, String queryPar
...
+    /**
+     * For endpoints that still allow either for transition, favors the query parameter as that's the likely user
+     * expectation since machine systems wouldn't said both.
+     * @param header content type from a header
+     * @param queryParam content type from a query parameter
+     * @param klass DTO to find a matching formatter for.
+     * @return ContentType appropriate to the given selection.
+     * @throws UnsupportedFormatException if there is no matching content type for the given class
+     */
+    public static ContentType parseQueryOrHeaderParam(String headerParam, String queryParam, Class<? extends CwmsDTOBase> klass) {
```
Comment (rma-rripken) [inline cwms-data-api/src/main/java/cwms/cda/formatters/Formats.java:292]: This needs unit tests to demo what this method is doing.

File: Dockerfile [hunk 2]
```diff
@@ -38,10 +40,12 @@ ENV CDA_POOL_INIT_SIZE "5"
...
 ENV CDA_POOL_MAX_IDLE "10"
 ENV CDA_POOL_MIN_IDLE "5"
-ENV cwms.dataapi.access.providers ""
-ENV cwms.dataapi.access.openid.wellKnownUrl ""
-ENV cwms.dataapi.access.openid.issuer ""
+ENV cwms.dataapi.access.providers "KeyAccessManager,OpenID"
+ENV cwms.dataapi.access.openid.wellKnownUrl "https://identity-test.cwbi.us/auth/realms/cwbi/.well-known/openid-configuration"
+ENV cwms.dataapi.access.openid.issuer "https://identity-test.cwbi.us/auth/realms/cwbi"
 ENV cwms.dataapi.access.openid.timeout "604800"
+ENV cwms.dataapi.access.openid.altAuthUrl "https://identityc-test.cwbi.us/auth/realms/cwbi"
```
Comment (rma-rripken) [inline Dockerfile:47]: Just confirming that you meant for these test.cwbi.us urls to be included.

**Review by rma-rripken — CHANGES_REQUESTED** [review]
Can you do these two things:

- [x] Add unit test on the new Formats method
- [x] Confirm you like those test.cwbi.us urls

File: Dockerfile (hunk 2 above)
Comment (MikeNeilson) [reply Dockerfile:47]: Yeah, may not be the best thing but at the moment I keep forgetting to update them and they aren't generally accessible. They will be changed to something generic again later.

**Review by rma-rripken — APPROVED** [review]

**Pull request merged**